tv_apps/browser/js/awesomescreen.js

```diff
@@ -3266,5 +3306,219 @@ var Awesomescreen = {
         .querySelector('.tab-content[data-content="' + dataContent +'"]')
         .classList.add('show');
     }
+  },
+
+  initSpatialNavigation: function awesomescreen_initSpatialNavigation() {
+    SpatialNavigation.init();
+
+    this.enableSpatialNavigation();
+
+    SpatialNavigation.makeFocusable();
+
+    SpatialNavigation.focus('tool-bar');
+
+    document.querySelector('.tab-navigation').addEventListener('sn:willmove',
+      function(e){
+        let target = e.target;
+        if (!target.classList.contains('tab-option')) {
+          return;
+        }
+
+        let detail = e.detail;
+        if (detail.direction === 'left' && !target.previousElementSibling) {
+          e.preventDefault();
+        }
+        if (detail.direction === 'right' && !target.nextElementSibling) {
+          e.preventDefault();
+        }
+      });
+
+    document.querySelector('#sync-tab-content .tab-list-view')
+      .addEventListener('keydown', function(e){
+        let firsListItem =
+          document
+            .querySelector(`#sync-tab-content
+              .tab-list-item:first-child .list-item`);
+
+        if (firsListItem !== e.target || e.keyCode !== KeyEvent.DOM_VK_UP) {
+          e.stopPropagation();
+        }
+      });
+  },
+
+  enableSpatialNavigation: function awesomescreen_disableSpatialNavigation() {
+    if (this.isSpatialNavigationEnable) {
+      return;
+    }
+
+    function getSearchBarStyle() {
+      if (Toolbar.toolbarPanel) {
+       return (Toolbar.getSearchBarStyle() == 'true');
+      } else {
+       let toolbarPanel = document.getElementById('toolbar-panel');
+       return (toolbarPanel.dataset.search == 'true');
+      }
+    }
+
+    let toolbarSelector = [
+      '#sidebar-button-block',
+      '#back-button-block',
+      '#address-form',
+      '#url-input',
+      '#search-form',
+      '#search-input',
+      '#search-button',
+      '#search-clear-button',
+      '#search-close-button',
+      '#show-bookmarks-button-block',
+      '#home-button-block',
+      '#tabs-button-block',
+      '#menu-button-block'
+    ];
+
+    SpatialNavigation.add({
+      id: 'tool-bar',
+      selector: toolbarSelector.join(','),
+      defaultElement: '#search-form',
+      enterTo: 'last-focused',
+      navigableFilter: function(el) {
```

Comment:
Although using `navigableFilter` is feasible, I wonder if it's possible to toggle some class, say "disabled", on elements when needed. Thus, you can set the selector like `#address-form:not(.disabled)` and avoid complex rules here.

Anyway, it's up to me.